### PR TITLE
Canonical URL banner

### LINKED ISSUE #115: Displayed plot size in tutorials
In the [visualization tutorials](https://scanpy-tutorials.readthedocs.io/en/latest/plotting/advanced.html) the plots are rendered to a box of predefined size/shape and plots are not properly displayed - they get the correct size only after clicking on the plots to open in a separate window. This is problematic for the tutorial as we want to show how to properly align/size plots and the current layout makes it seem like plots are not properly sized. 
See the example below that is completely squashed:
image
While the image opened in a separate tab looks normal:
image

### PR DESCRIPTION
**Pull request by flying-sheep**
Fixes #115

Also fixes all warnings

To review: Just click through the tutorials here: https://icb-scanpy-tutorials--116.com.readthedocs.build/en/116/

and tell me if the big boxes on top look helpful enough or if we need more drastic measures

### CODE REVIEW

File: conf.py [hunk 1]
```diff
@@ -48,30 +65,54 @@
...
-        return node_list
-    return [image]
+    def run(self) -> list[nodes.Node]:
+        text = self.arguments[0]
+        ref = resolve_reference_in_inventory(
+            self.env,
+            "scanpy",
+            addnodes.pending_xref("", reftype="doc", refdomain="std", reftarget=text),
+            nodes.inline("", text),
+        )
```
Comment (ilan-gold) [inline conf.py:104]: I can't find this on the `intersphinx` docs page so I really can't say what it does: https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#module-sphinx.ext.intersphinx also googling gave me nothing

File: conf.py [hunk 2]
```diff
@@ -48,30 +65,54 @@
...
+    content: Sequence[str] = (),
+) -> tuple[list[nodes.Node], list[str]]:
+    node = inliner.document.reporter.info(f"cite:{text}")
+    return [node], []
+
+
+class FakeDomain(Domain):
+    name = "cite"
+    roles = dict(p=fake_cite, t=fake_cite)
+
```
Comment (ilan-gold) [inline conf.py:88]: Is this for the warnings?  What does it do?

Comment (flying-sheep) [reply conf.py:88]: What it says in the comment in line 69: It makes ``{cite}`Foo17` `` and ``{cite:p}`Bar22` `` not crash the docs build.

File: conf.py (hunk 1 above)
Comment (flying-sheep) [reply conf.py:104]: This makes this

```
:::{canonical-tutorial} tutorials/experimental/dask
:::
```

render into this:

image

File: conf.py (hunk 2 above)
Comment (flying-sheep) [reply conf.py]: But good point, I changed it so it actually shows the citation text

File: conf.py (hunk 1 above)
Comment (flying-sheep) [reply conf.py:107]: the function can return `None` which I catch with the `assert`. So the docs don’t build if one of these tries to link with a doc that isn’t in scanpy.

Comment (ilan-gold) [reply conf.py:107]: I think my point is that a comment might be nice so that the next person can know what with function is doing.

**Comment by Hrovatin** [comment]
If users are advised to redirect to the other page in the message pasted below, I would make that more clear, e.g. "Please access this document in its original location as the currently accessed page may not be rendered correctly." or sth like this
image

**Comment by ilan-gold** [comment]
+1 for Karin's suggestion

File: conf.py (hunk 1 above)
Comment (flying-sheep) [reply conf.py:107]: The docstring helps, but sure.

grafik

**Comment by flying-sheep** [comment]
Text changed, and I added a spooky-looking striped background

**Pull request merged**